modules/varnish/templates/default.vcl

```diff
@@ -346,6 +348,17 @@ sub vcl_hash {
 	}
 }
 
+sub vcl_hit {
+	if (!obj.ttl > 0s) {
+		return (pass);
+	}
+
+	# Force lookup if the request is a no-cache request from the client.
+	if (req.http.Cache-Control ~ "no-cache") {
+		return (miss);
+	}
+}
+

```

Comment:
What if neither if is executed? This would abandon the request entirely rather than return a result.